pkg/cmd/cli/cmd/env.go

```diff
@@ -371,7 +384,11 @@ func updateEnv(existing []kapi.EnvVar, env []kapi.EnvVar, remove []string) []kap
 		covered.Insert(e.Name)
 		out = append(out, e)
 	}
-	return out
+	envUpdated := true
```

Comment:
Then the line which would be setting `envUpdate` to false would need to be basically in all three `if` statements and also at the end of the `for` loops, due to the logic the `out` variable is build. Would rather prevent prevent that.